extractor/extractor.cpp

```diff
@@ -253,6 +262,57 @@ int extractor::run()
 
         TIMER_STOP(extracting);
         SimpleLogger().Write() << "extraction finished after " << TIMER_SEC(extracting) << "s";
+    }
+    catch (std::exception &e)
+    {
+        SimpleLogger().Write(logWARNING) << e.what();
+        return 1;
+    }
+
+    try
+    {
+        // Transform the node-based graph that OSM is based on into an edge-based graph
+        // that is better for routing.  Every edge becomes a node, and every valid
+        // movement (e.g. turn from A->B, and B->A) becomes an edge
+        //
+        //
+        //    // Create a new lua state
+
+        SimpleLogger().Write() << "Generating edge-expanded graph representation";
+
+        TIMER_START(expansion);
+
+        auto node_based_edge_list = osrm::make_unique<std::vector<EdgeBasedNode>>();;
```

Comment:
I fixed this and similar issues in https://github.com/Project-OSRM/osrm-backend/pull/1664. Please don't introduce them back into the code base.

`std::vector` already owns and thus manages its resources. There is no need for `std::unique_ptr<std::vector<>>`, as this brings no benefits, but adds one level of indirection. If you want to have a read-only view of a vector, take it by const-ref. If you take ownership of a vector, take it by-value leaving the call site with the choice to either pass in a copy or use `std::move` to transfer ownership.